src/env/obfuscated.rs

```diff
@@ -84,11 +86,54 @@ impl ObfuscatedFileSystem {
         self.files.load(Ordering::Relaxed)
     }
 }
+pub struct ObfuscatedContext(<DefaultFileSystem as FileSystem>::AsyncIoContext);
+impl AsyncContext for ObfuscatedContext {
+    fn wait(&mut self) -> IoResult<usize> {
+        self.0.wait()
+    }
+
+    fn data(&self, seq: usize) -> &[u8] {
+        self.0.data(seq)
+    }
+
+    fn single_wait(&mut self, seq: usize) -> IoResult<usize> {
+        self.0.single_wait(seq)
+    }
+
+    fn submit_read_req(&mut self, buf: Vec<u8>, offset: u64) -> IoResult<()> {
+        self.0.submit_read_req(buf, offset)
+    }
+}
 
 impl FileSystem for ObfuscatedFileSystem {
     type Handle = <DefaultFileSystem as FileSystem>::Handle;
     type Reader = ObfuscatedReader;
     type Writer = ObfuscatedWriter;
+    type AsyncIoContext = ObfuscatedContext;
```

Comment:
Why? Doesn't this work already?
`type AsyncIoContext = <DefaultFileSystem as FileSystem>::AsyncIoContext`

Reply:
@tabokie I introduced the `ObfuscatedContext` struct based on your comments:https://github.com/tikv/raft-engine/pull/286#discussion_r1058024065

Reply:
Adding a layer of wrapper is only useful if you have implemented something on top of the base context. Right now you didn't implement anything, so there is no need to wrap it.